src/aggregator/diff-aggregator.ts

```diff
@@ -52,45 +52,47 @@ export class DiffAggregator implements Aggregator {
     validateNonNull(this.adapter, 'Missing destination adapter');
 
     const exportResult = await this.adapter!.exportAllEntities();
-    const storedItems = this.normalize(exportResult.importAction);
-    const collectedItems = this.normalize(externalContent);
+    const {
+      categories: storedCategories = [],
+      labels: storedLabels = [],
+      documents: storedDocuments = [],
+    } = this.removeGeneratedContent(exportResult.importAction);
+    const {
+      categories: collectedCategories = [],
+      labels: collectedLabels = [],
+      documents: collectedDocuments = [],
+    } = externalContent;
+
+    this.resolveNameConflicts(collectedCategories, storedCategories);
+    this.resolveNameConflicts(collectedLabels, storedLabels);
+
+    const categories = this.collectModifiedItems(
+      collectedCategories,
+      storedCategories,
+      this.normalizeCategory.bind(this),
+    );
 
-    this.resolveNameConflicts(
-      collectedItems.categories,
-      storedItems.categories,
+    const labels = this.collectModifiedItems(
+      collectedLabels,
+      storedLabels,
+      this.normalizeLabel.bind(this),
     );
-    this.resolveNameConflicts(collectedItems.labels, storedItems.labels);
-    this.removeGeneratedContent(storedItems);
 
-    const importAction = exportResult.importAction;
-    return {
-      categories: this.collectModifiedItems(
-        collectedItems.categories,
-        importAction.categories || [],
-        this.normalizeCategory.bind(this),
-      ),
-      labels: this.collectModifiedItems(
-        collectedItems.labels,
-        importAction.labels || [],
-        this.normalizeLabel.bind(this),
-      ),
-      documents: this.collectModifiedItems(
-        collectedItems.documents,
-        importAction.documents || [],
-        this.normalizeDocument.bind(this),
-      ),
-    };
-  }
+    const documents = this.collectModifiedItems(
+      collectedDocuments,
+      storedDocuments,
+      (doc) =>
+        this.normalizeDocument(
```

Comment:
The main changes is here to send in the resolved category and label list. Document normalizer can look up categories and labels from these lists and can use the non-conflicting name of them.